applications/soupbintcp-gateway/src/main/java/com/paritytrading/nassau/soupbintcp/gateway/Events.java

```diff
@@ -106,13 +106,8 @@ public void receive() throws IOException {
                 continue;
 
             for (int i = 0; i < toCleanUp.size(); i++) {
-                Session session = toCleanUp.get(i);
-
-                toKeepAlive.remove(session);
-
-                try {
-                    session.close();
-                } catch (IOException e) {
+                try (Session session = toCleanUp.get(i)) {
+                    toKeepAlive.remove(session);
```

Comment:
This change took a while to grow on me, but now I think it's definitely better than before. 👍